shared-functions.sh

```diff
@@ -81,14 +81,14 @@ ensure_mac_os() {
 # Mac-specific function to install Java JDK
 install_mac_java() {
     # Determine which Java JDKs we have.
-    # If a non-Adopt Open JDK that is v1.8 (aka Java 8), that's ok!
-    java_versions=$(/usr/libexec/java_home --version "1.8" >/dev/null 2>&1 || echo "Not found")
+    # If a non-Adopt Open JDK that is v1.11 (aka Java 11), that's ok!
+    java_versions=$(/usr/libexec/java_home --version "1.11" >/dev/null 2>&1 || echo "Not found")
```

Comment:
Is there a command we can use instead?  I would like to get this right since I'd like to tell people to just rerun dotfiles to get it.

Reply:
I think this thing hinges on how important we think `If a non-Adopt Open JDK that is v1.11 (aka Java 11), that's ok!` is to us.

The simplest thing to me seems to always brew install.  If `adoptopenjdk11` is already installed, brew will tell you so without blocking the script.  Folks don't run the dot files often enough to make this a performance concern, me thinks.  As you already multiple JDKs can coexist so its a matter of correctly setting `$JAVA_HOME`.

But if we wanted to potentially improve our version detection logic we can clobber something together.  If you used `/usr/libexec/java_home -V` you can tell all the versions you have installed and check if 11 is there.  This is sample of the output

```
➜  khan /usr/libexec/java_home -V              
Matching Java Virtual Machines (2):
    11.0.11 (x86_64) "AdoptOpenJDK" - "AdoptOpenJDK 11" /Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
    1.8.0_292 (x86_64) "AdoptOpenJDK" - "AdoptOpenJDK 8" /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home
/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
````

We can determine from that if `AdoptOpenJDK 11` is installed.  We'd want to check if `$JAVA_HOME` is set to the version we want.  At this point we can determine what to do; set `$JAVA_HOME` to the new version and override what's already or leave it alone if its already set to `AdoptOpenJDK 11`.   We can prompt engineers if they want to override `$JAVA_HOME` if its not `AdoptOpenJDK 11`...?

Just some thoughts.

Reply:
I think another thing we could tweak here, instead of using `/usr/libexec/java_home -V` is to test if `/usr/libexec/java_home --version "1.11"` contains `adoptopenjdk-11` in it instead of checking if nothing is returned, which is where the weak spot is in the current code.

When I have `adoptopenjdk11` installed, this is what I get:

```
➜  khan /usr/libexec/java_home --version "1.11"
/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
```

When I only have `adoptopenjdk8`, this is what I get (which is the issue Im mentioning here):
```
➜  khan /usr/libexec/java_home --version "1.11"
/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home
```